flink-core/src/main/java/org/apache/flink/metrics/MetricRegistry.java

```diff
@@ -85,7 +87,14 @@ public MetricRegistry(Configuration config) {
 		if (className == null) {
 			// by default, create JMX metrics
 			LOG.info("No metrics reporter configured, exposing metrics via JMX");
+			
+			Configuration reporterConfig = new Configuration();
+			String portRange = config.getString(KEY_METRICS_JMX_PORT, null);
+			if (portRange != null) {
+				reporterConfig.setString(KEY_METRICS_JMX_PORT, portRange);
+			}
 			this.reporter = new JMXReporter();
+			this.reporter.open(reporterConfig);
```

Comment:
There is also the catch clause of the else branch where we create a `JMXReporter`, but we never call the `open` method there. I think we should add a proper `open` call there as well.